nilearn/reporting/_plot_matrices.py

```diff
@@ -62,6 +64,94 @@ def plot_design_matrix(design_matrix, rescale=True, ax=None, output_file=None):
     return ax
 
 
+def plot_event(model_event, cmap=None, output_file=None, **fig_kwargs):
+    """Creates plot for event visualization.
+
+    Parameters
+    ----------
+    model_event : pandas DataFrame or list of pandas DataFrame
+        the `pandas.DataFrame` must have three columns
+        ``event_type`` with event name, ``onset`` and ``duration``.
+        The `pandas.DataFrame` can also be obtained from 
+        :func:`~nilearn.stats.first_level_model.first_level_models_from_bids`.
+
+    cmap : str or matplotlib.cmap, optional
+        the colormap used to label different events
+
+    output_file : string or None, optional,
+        The name of an image file to export the plot to. Valid extensions
+        are .png, .pdf, .svg. If output_file is not None, the plot
+        is saved to a file, and the display is closed.
+
+    **fig_kwargs : extra keyword arguments, optional
+        Extra arguments passed to matplotlib.pyplot.subplots
+
+    Returns
+    -------
+    Plot Figure object
+
+    """
+
+    if isinstance(model_event, pd.DataFrame):
+        model_event = [model_event]
+   
+    n_runs = len(model_event)
+    figure, ax = plt.subplots(1, 1, **fig_kwargs)
+
+    # input validation
+    if cmap is None:
+        cmap = plt.cm.tab20
+    elif isinstance(cmap, str):
+        cmap = plt.get_cmap(cmap)
+    else:
+        cmap = cmap
+
+    event_labels = pd.concat(e['trial_type'] for e in model_event)
+    event_labels = np.unique(event_labels)
+
+    cmap_dictionary = {label:idx for idx, label in enumerate(event_labels)}
+
+    if len(event_labels) > cmap.N:
+        plt.close()
+        raise ValueError("The number of event types is greater than "+ \
+            " colors in colormap (%d > %d). Use a different colormap." \
+            % (len(event_labels), cmap.N))
+
+    for idx_run, event_df in enumerate(model_event):
+        
+        for _, event in event_df.iterrows():
+            event_onset = event['onset']
+            event_end = event['onset'] + event['duration']
+            color = cmap.colors[cmap_dictionary[event['trial_type']]]
+         
+            ax.axvspan(event_onset, 
+                       event_end, 
+                       ymin=(idx_run + .25) / n_runs, 
+                       ymax=(idx_run + .75) / n_runs, 
+                       facecolor=color)
+
+    handles = []
+    for label, idx in cmap_dictionary.items():
+        patch = mpatches.Patch(color=cmap.colors[idx], label=label)
+        handles.append(patch)
+
+    _ = ax.legend(handles=handles)
```

Comment:
As visible on https://5706-1235740-gh.circle-artifacts.com/0/doc/_build/html/auto_examples/04_glm_first_level_models/plot_events_file.html#sphx-glr-auto-examples-04-glm-first-level-models-plot-events-file-py
I think that the plot would be nicer if you increased a bit the number of cols in the legend, using "ncols=3" or 4.

Reply:
Ok!